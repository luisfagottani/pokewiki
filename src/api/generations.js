import { fetchApi } from './api';

const Generations = {
  getAllGenerations: async () => {
    return fetchApi(`generation`, {
      method: 'GET',
    });
  },

  getGenerationsById: async (id) => {
    return fetchApi(`generation/${id}`, {
      method: 'GET',
    });
  },
};

export default Generations;
