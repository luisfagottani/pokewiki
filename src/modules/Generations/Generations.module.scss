.generations {
  // Box Model
  display: flex;
  flex-wrap: wrap;
  margin: 0px -10px;

  &__item {
    // Box Model
    flex-basis: 50%;
    padding: 0 10px;
    margin-bottom: 20px;

    // Visual
    overflow: hidden;
  }

  &__content {
    // Box Model
    width: 100%;
    display: block;
    border: 1px solid var(--iron);
    border-radius: 4px;
    padding: 20px;
    display: flex;
    align-items: center;
    justify-content: center;

    // Typography
    text-decoration: none;

    // Visual
    transition: all ease-in-out 0.2s;

    &:hover {
      // Visual
      box-shadow: #28282880 0px 0px 25px -12px;
      transition: all ease-in-out 0.2s;
    }
  }

  &__label {
    // Typography
    text-align: center;
    font-weight: 600;
    color: var(--azure);
    text-decoration: none;
    text-transform: uppercase;
  }
}

@media screen and (min-width: 700px) {
  .generations {
    &__item {
      // Box Model
      flex-basis: 33%;
    }
  }
}

@media screen and (min-width: 1024px) {
  .generations {
    &__item {
      // Box Model
      // Box Model
      flex-basis: 25%;
    }
  }
}
