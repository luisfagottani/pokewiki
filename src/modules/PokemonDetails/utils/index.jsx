const progressBarColor = (progress) => {
  if (progress < 60) {
    return 'blue';
  } else if (progress >= 60 && progress <= 70) {
    return 'yellow';
  } else {
    return 'red';
  }
};

export { progressBarColor };
