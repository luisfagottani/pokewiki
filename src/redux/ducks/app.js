import produce from 'immer';

export const REDUCER_NAME = 'app';

// TYPES
export const Types = {
  SET_GLOBAL_LOADING: `${REDUCER_NAME}/SET_GLOBAL_LOADING`,
};

// INITIAL STATE
const initialState = {
  isLoading: true,
  locale: 'pt-BR',
};

// REDUCER
export const reducer = produce((draft = initialState, action) => {
  switch (action.type) {
    case Types.SET_GLOBAL_LOADING:
      draft.isLoading = action.payload;
      return draft;

    case Types.SET_USER_PROFILE:
      draft.profile = action.profile;
      return draft;

    default:
      return draft;
  }
});

// ACTION FUNCTIONS

export function defineGlobalLoading(status) {
  return {
    type: Types.SET_GLOBAL_LOADING,
    payload: status,
  };
}

// SELECTOR FUNCTIONS
export function getGlobalLoading(state) {
  return state[REDUCER_NAME].isLoading;
}

export function getLocale(state) {
  return state[REDUCER_NAME].locale;
}
