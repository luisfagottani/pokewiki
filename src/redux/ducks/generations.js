import produce from 'immer';

export const REDUCER_NAME = 'family';

// TYPES
export const Types = {
  FETCH_GENERATIONS: `${REDUCER_NAME}/FETCH_POKEMONS`,
  FETCH_GENERATIONS_BY_ID: `${REDUCER_NAME}/FETCH_GENERATIONS_BY_ID`,
  PERSIST_GENERATIONS: `${REDUCER_NAME}/PERSIST_GENERATIONS`,
};

// INITIAL STATE
const initialState = {
  generations: [],
};

// REDUCER
export const reducer = produce((draft = initialState, action) => {
  switch (action.type) {
    case Types.PERSIST_GENERATIONS:
      draft.generations = { ...action.payload };
      return draft;

    default:
      return draft;
  }
});

// ACTION FUNCTIONS
export function fetchGenerations() {
  return {
    type: Types.FETCH_GENERATIONS,
  };
}

export function fetchGenerationsById(id) {
  return {
    type: Types.FETCH_GENERATIONS_BY_ID,
    payload: id,
  };
}

export function persistGenerations(pokemons) {
  return {
    type: Types.PERSIST_GENERATIONS,
    payload: pokemons,
  };
}

// SELECTORS
export function getGenerations(state) {
  return state[REDUCER_NAME]?.generations;
}
