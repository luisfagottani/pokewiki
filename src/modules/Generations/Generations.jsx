import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { getGenerations } from 'redux/ducks/generations';

import style from './Generations.module.scss';

const Generations = ({ generations = [] }) => (
  <ul className={style.generations}>
    {Object.values(generations).map((generation, key) => (
      <li key={generation?.name} className={style.generations__item}>
        <Link to={`generations/${generation?.name}`} className={style.generations__content}>
          <span className={style.generations__label}>{generation?.name}</span>
        </Link>
      </li>
    ))}
  </ul>
);

const mapStateToProps = (state) => ({
  generations: getGenerations(state),
});

export default connect(mapStateToProps)(Generations);
