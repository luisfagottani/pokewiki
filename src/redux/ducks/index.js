import { combineReducers } from 'redux';

import { reducer as AppReducer, REDUCER_NAME as APP_REDUCER_NAME } from './app';
import { reducer as PokemonsReducer, REDUCER_NAME as APP_POKEMONS_REDUCER } from './pokemons';
import { reducer as GenerationsReducer, REDUCER_NAME as APP_GENERATIONS_REDUCER } from './generations';

export default combineReducers({
  [APP_REDUCER_NAME]: AppReducer,
  [APP_POKEMONS_REDUCER]: PokemonsReducer,
  [APP_GENERATIONS_REDUCER]: GenerationsReducer,
});
