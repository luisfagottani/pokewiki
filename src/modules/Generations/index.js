import Generations from './Generations';

export default Generations;
