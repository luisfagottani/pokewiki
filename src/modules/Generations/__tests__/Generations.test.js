import React from 'react';
import { createMockStore, render } from 'utils/tests';
import { reducer as AppReducer, REDUCER_NAME as APP_REDUCER_NAME } from 'redux/ducks/app';
import { reducer as GenerationReducer, REDUCER_NAME as GENERATION_REDUCER_NAME } from 'redux/ducks/generations';
import Generations from '../';

const mockStore = createMockStore({
  reducers: { [APP_REDUCER_NAME]: AppReducer, [GENERATION_REDUCER_NAME]: GenerationReducer },
  initialState: {
    [APP_REDUCER_NAME]: {
      isLoading: true,
      locale: 'EN',
    },
    [GENERATION_REDUCER_NAME]: {
      generations: [{ name: 'generation-i' }, { name: 'generation-ii' }, { name: 'generation-iii' }],
    },
  },
});

describe('<Generations />', () => {
  it('Should render without errors', () => {
    const { container } = render(<Generations />, mockStore);
    expect(container).toMatchSnapshot();
  });

  it('should render with the correct generations', () => {
    const { getByText } = render(<Generations />, mockStore);

    getByText('generation-i');
    getByText('generation-ii');
    getByText('generation-iii');
  });
});
