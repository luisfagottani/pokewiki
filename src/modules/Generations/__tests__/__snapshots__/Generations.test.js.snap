// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Generations /> Should render without errors 1`] = `
<div>
  <ul
    class="generations"
  >
    <li
      class="generations__item"
    >
      <a
        class="generations__content"
        href="/generations/generation-i"
      >
        <span
          class="generations__label"
        >
          generation-i
        </span>
      </a>
    </li>
    <li
      class="generations__item"
    >
      <a
        class="generations__content"
        href="/generations/generation-ii"
      >
        <span
          class="generations__label"
        >
          generation-ii
        </span>
      </a>
    </li>
    <li
      class="generations__item"
    >
      <a
        class="generations__content"
        href="/generations/generation-iii"
      >
        <span
          class="generations__label"
        >
          generation-iii
        </span>
      </a>
    </li>
  </ul>
</div>
`;
